general/networking/nginx.md

```diff
@@ -10,65 +10,87 @@ title: Nginx Reverse Proxy
 Create the file `/etc/nginx/conf.d/jellyfin.conf` which will forward requests to Jellyfin.
 
 ```
+# Uncomment the commented sections after you have acquired a SSL Certificate
 server {
     listen 80;
-    server_name DOMAIN_NAME;
+    listen [::]:80;
+    # server_name DOMAIN_NAME;
 
     # Uncomment to redirect HTTP to HTTPS
     # return 301 https://$host$request_uri;
-}
+#}
 
-# Uncomment this section after you have acquired a SSL Certificate
 #server {
-#    listen 443 ssl http2;
-#    server_name DOMAIN_NAME;
-#    ssl_certificate /etc/letsencrypt/live/DOMAIN_NAME/fullchain.pem;
-#    ssl_certificate_key /etc/letsencrypt/live/DOMAIN_NAME/privkey.pem;
-#    include /etc/letsencrypt/options-ssl-nginx.conf;
-#    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
-#    add_header Strict-Transport-Security "max-age=31536000" always;
-#    ssl_trusted_certificate /etc/letsencrypt/live/DOMAIN_NAME/chain.pem;
-#    ssl_stapling on;
-#    ssl_stapling_verify on;
-#
-#    # Security / XSS Mitigation Headers
-#    add_header X-Frame-Options "SAMEORIGIN";
-#    add_header X-XSS-Protection "1; mode=block";
-#    add_header X-Content-Type-Options "nosniff";
-#
-#    # Content Security Policy
-#    # See: https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP
-#    # Enforces https content and restricts JS/CSS to origin
-#    # External Javascript (such as cast_sender.js for Chromecast) must be whitelisted.
-#    add_header Content-Security-Policy "default-src https: data: blob:; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline' https://www.gstatic.com/cv/js/sender/v1/cast_sender.js; worker-src 'self' blob:; connect-src 'self'; object-src 'none'; frame-ancestors 'self'";
-#
-#    location / {
-#        # Proxy main Jellyfin traffic
-#        proxy_pass http://SERVER_IP_ADDRESS:8096/;
-#        proxy_set_header Host $host;
-#        proxy_set_header X-Real-IP $remote_addr;
-#        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-#        proxy_set_header X-Forwarded-Proto $scheme;
-#        proxy_set_header X-Forwarded-Protocol $scheme;
-#        proxy_set_header X-Forwarded-Host $http_host;
-#
-#        # Disable buffering when the nginx proxy gets very resource heavy upon streaming
-#        proxy_buffering off;
-#    }
-#    location /socket {
-#        # Proxy Jellyfin Websockets traffic
-#        proxy_pass http://SERVER_IP_ADDRESS:8096/socket;
-#        proxy_http_version 1.1;
-#        proxy_set_header Upgrade $http_upgrade;
-#        proxy_set_header Connection "upgrade";
-#        proxy_set_header Host $host;
-#        proxy_set_header X-Real-IP $remote_addr;
-#        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
-#        proxy_set_header X-Forwarded-Proto $scheme;
-#        proxy_set_header X-Forwarded-Protocol $scheme;
-#        proxy_set_header X-Forwarded-Host $http_host;
-#    }
-#}
+    # listen 443 ssl http2;
```

Comment:
Haven't had a chance to test yet (afk until tomorrow, sorry) but I personally don't like this change to the comments.  If we're going to have large blocks commented out, I'd prefer them all to be in a consistent line so they can easily be removed in an editor.  having them staggered and indented means a lot of manual work.  Plus it's commented out here, but uncommented out at L54.  The mix and match doesn't make sense.  

Previously in chat we'd discussed removing the commented out lines entirely and just having 2 blocks, one for 80 and one for 443 without doing the "uncomment this for SSL" thing.  I don't remember the result of that discussion though.  @Artiume, do you?

Reply:
With finding a lot of people on reddit exposing their servers, it might be better to give warnings of using http, provide a copypasta config and hope they use it to learn only. I'd like to make it a narrative of basic, advanced, more advanced.